### ISSUE TITLE
Display of closed captions (VTT) settings in media player

### ISSUE THREAD

**Issue by edsu** [sul-dlss/sul-embed#1498]
### Expected behavior

VTT closed captions settings are styled correctly. See #1469 for context.

### Actual behavior

The caption settings are not displaying correctly:

<img width="1230" alt="Screenshot 2023-05-17 at 2 07 33 PM" src="https://github.com/sul-dlss/sul-embed/assets/92044/5e218cef-b3c1-4186-95e8-9c966a2d8c18">

It seems like the .vjs-modal-dialog styles are not being applied?

There's also a CORS error:
```
Security Error: Content at https://embed-stage.stanford.edu/iframe?url=https://sul-purl-stage.stanford.edu/dw202dq8174&_v=1684274192 may not load data from https://sul-stacks-stage.stanford.edu/file/druid:dw202dq8174/Redivis_unedited_GMT20220303-205959_Recording.transcript.vtt.
```

### Steps to reproduce the behavior

Visit https://sul-purl-stage.stanford.edu/dw202dq8174 and click to play the video. You should see a CC icon appear in the lower right area of the video playback component. If you click that and then "Caption Settings" you should see the settings modal display.

### Browser / Environment

Firefox / AppleOS

**Comment by dinahhandel** [comment]
### Expected behavior

A user should be able to choose if they want captions to display by clicking a button or otherwise selecting the functionality.
The CC icon as well as caption settings options must display in a way that does not interfere with playback of the media (as dropdown menus? As a pop-up over the media?).

### Other considerations

What if there is no caption file? Should an option to play CC still display? Is it possible for a viewer to only show an option to play CC if there is a web.vtt file present? Or is CC hard-coded into the viewer and we need to notify a user that if there is no web.vtt file, captions are not yet available for playback? If this is the case, can we build in better functionality for requesting that the media be captioned?

**Comment by dinahhandel** [comment]
@jcoyne I'm starting to feel like this ticket doesn't accurately portray the desired outcome. Is the preferred method to add onto this ticket, or start a new one? To me, this ticket is about how the caption display settings function, and doesn't encompass the functionality around caption/transcript display.

**Comment by jcoyne** [comment]
@dinahhandel I prefer we just add onto this on. I find it hard to have a multitude of tickets that seem to all be referring to the same thing.

**Comment by edsu** [comment]
@dinahhandel this may not need saying but please feel free to modify the initial description if it helps.

**Comment by jcoyne** [comment]
From @dinahhandel on #1469
Expected behavior

A user should be able to choose if they want captions to display by clicking a button or otherwise selecting the functionality.
The CC icon as well as caption settings options must display in a way that does not interfere with playback of the media (as dropdown menus? As a pop-up over the media?).
Other considerations

What if there is no caption file? Should an option to play CC still display? Is it possible for a viewer to only show an option to play CC if there is a web.vtt file present? Or is CC hard-coded into the viewer and we need to notify a user that if there is no web.vtt file, captions are not yet available for playback? If this is the case, can we build in better functionality for requesting that the media be captioned? What about multi-lingual caption support?

**Comment by mjgiarlo** [comment]
Question: do we want to have closed caption settings at all? It would help to see visual examples to decide.

**Comment by dinahhandel** [comment]
This is a nice to have, but not off the table entirely.

**Comment by mjgiarlo** [comment]
Thanks for re-opening!

**Comment by edsu** [comment]
I think this should be closed now that #1602 has been merged?

**Comment by jcoyne** [comment]
@edsu no, I think we still want this to reenable the settings (and style them correctly)

**Comment by dinahhandel** [comment]
Ideally, the user would open a modal that covers the viewer to select caption display settings. They could also exit the modal in the same way other modals are exited in the media player.

**Comment by alundgard** [comment]
See the "Single Companion Window Desktop" prototypes in the [Figma Design](https://www.figma.com/file/ADNitlk4aWHDiPDWD1dHdF/Media-Work-Cycle---Media-Player?type=design&node-id=0-1&mode=design
). [Click the "Present" play arrow in the top-right corner and select these prototypes.]

![Image](https://github.com/sul-dlss/sul-embed/assets/2343316/f8a20fb2-34e5-44b3-aab0-2822b0156b87)

Clicking the gear icon in the video controls opens a Settings modal. Caption settings are selected from dropdown menus. The user can Save settings, or Close to dismiss the modal without saving.

Nice to have: a "Sample caption text" preview of how the settings changes will appear.

**Comment by aaron-collier** [comment]
This was partially addressed in #1764, however applying the full design for the settings window is out of scope as the code is not local and instead served from the player. Some CSS work was done to clean up the view, but changing/moving components in the design is potentially problematic and could require some funky JS hacking.
